pkg/pillar/cmd/domainmgr/sriov.go

```diff
@@ -0,0 +1,84 @@
+package domainmgr
+
+import (
+	"fmt"
+	"net"
+	"os"
+	"strconv"
+	"time"
+
+	"github.com/lf-edge/eve/pkg/pillar/types"
+	"github.com/vishvananda/netlink"
+)
+
+func vfCount(ib types.PhysicalIOAdapter) (uint8, error) {
+	v, exists := ib.Cbattr[types.VfCountFieldName]
+
+	if exists {
+		c, err := strconv.Atoi(v)
+		if err != nil {
+			return 0, fmt.Errorf("Cannot convert %s to integer. Value %s, error: %v", types.VfCountFieldName, v, err)
+		} else {
+			return uint8(c), nil
+		}
+	} else {
+		return 0, nil
+	}
+}
+
+func createVF(device string, vfCount uint8) error {
+	f := fmt.Sprint(types.NicLinuxPath, device, types.NumvfsDevicePath)
```

Comment:
Does this sprintf work? You just want to concatenate three parts of the pathname which can be done with string concatenation if you manually add the "/" before and after device, or using the path/filepath package.

Reply:
used filepath library